docs/literate/src/files/subcell_shock_capturing.jl

```diff
@@ -108,6 +109,11 @@ positivity_variables_nonlinear = [pressure]
 # the following.
 local_twosided_variables_cons = ["rho"]
 
+# To limit non-linear variables locally, pass the variable function combined with the requested
+# bound (`min` or `max`) as a tuple. For instance, to impose a lower local bound on the modified
+# specific entropy by Guermond et al. use
+local_onesided_variables_nonlinear = [(Trixi.entropy_guermond_etal, min)]
```

Comment:
I would suggest making `entropy_guermond_etal` public and document it with an appropriate docstring.
Then, you could link to that docstring using `@ref`.

Reply:
Does it make sense to make this specific entropy public, and all others we have implemented not? Especially, when the definitions differ for each type of equations.
I'm not sure about that.

Reply:
I did not check if there are different entropies implemented, sorry.
Is the scenario depicted here an exception, or is it typical to specify this entropy as a limited variable? In the latter we should maybe export it anyways for consistency.

For this PR it is probably sufficient to provide a link to the paper where this specific entropy is introduced.

Reply:
Okay sorry. There are not that many. I was mistaken. For instance, for compressible Euler 2D there is mathematical entropy (which I'm using for subcell limiting as well) and thermodynamic (which is used in the calculation for the mathematical).
I still think it's sufficient to just add the link here in the tutorial. No real need to overdo.

Reply:
Or what do @sloede and @amrueda think?

Reply:
I think it would be good to add the link to the paper in the tutorial. I saw that the paper is already referenced in a comment in the header of the function `entropy_guermond_etal`.

Reply:
I don't have anything against making the entropy functions public, but maybe this PR is not the place for that.

Reply:
Naive question: do we only add docstrings to exported functions? I mean, isn't it possible to fetch the docstring by doing, .e.g.,
```julia
?Trixi.entropy_guermond_etal
```
If that's the case (i believe it is), I would convert those header comments into docstrings...

Reply:
AFAIK, Julia never had a clear-cut notion of how to identify what's part of the API of a package and what isn't. Therefore, we used our own interpretation (one shared with many other packages) that anything that is `export`ed and/or has a docstring is part of our API. Part of the API means that we will not introduce a breaking change to such a name unless in a breaking release. 

With Julia v1.11 we will get the `public` keyword to indicate that something is part of the API even of it is not exported. Thus, once we make 1.11 the base supported version, we will be able to make this distinction explicitly.

For now, I'll leave it up to you. Personally, I don't think that such a specialized flux needs to be part of the API. However, more documentation is good, and since users *will* have to interact with it anyways, there's no harm to add a docstring. I don't think it needs to be exported (yet), though.

Reply:
I just tried using the doctring here. Is it expected that even with the following doctring
```
"""
    entropy_guermond_etal(u, equations::CompressibleEulerEquations2D)

Calculate the modified specific entropy of Guermond et al. (2019):
s_0 = p * rho^(-gamma) / (gamma-1).
Note: This is *not* the "conventional" specific entropy s = ln(p / rho^(gamma)).
- Guermond at al. (2019)
  Invariant domain preserving discretization-independent schemes and convex limiting for hyperbolic systems.
  [DOI: 10.1016/j.cma.2018.11.036](https://doi.org/10.1016/j.cma.2018.11.036)
"""
```
in the tutorial `[`entropy_guermond_etal`](@ref)` was not known and making the docs failed (the routine itself is not exported)?

So, I just linked the paper within the tutorial now.

Reply:
Maybe try ```[`Trixi.entropy_guermond_etal`](@ref)```?

Reply:
Thanks. That worked. Pushed it in [11303c3](https://github.com/trixi-framework/Trixi.jl/pull/1934/commits/11303c34abf41c370d01e163eb879dfbfb6a6f7c).